collect_app/src/main/java/org/odk/collect/android/http/HttpInterface.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2018 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.odk.collect.android.http;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.odk.collect.android.utilities.ResponseMessageParser;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+public interface HttpInterface {
+
+    /**
+     * Creates a http connection and sets up an input stream.
+     *
+     * @param uri of the stream
+     * @param contentType check the returned Mime Type to ensure it matches. "text/xml" causes a Hash to be calculated
+     * @return HttpInputStreamResult - An object containing the Stream, Hash and Headers
+     * @throws Exception a multitude of Exceptions such as IOException can be thrown
+     */
+    @NonNull
+    HttpInputStreamResult getHttpInputStream(@NonNull URI uri, @Nullable String contentType) throws Exception;
```

Comment:
So, this method executes an HTTP GET request to the given URI and it enforces the given content type if it's defined and returns a result object. Why not rename the method to `get(@NonNull URI uri, @Nullable String expectedContentType)` and make it return an `HttpGetResult` object?